pkg/kube/controllers/boshdeployment/reconciler.go

```diff
@@ -108,6 +164,200 @@ func (r *ReconcileBOSHDeployment) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, errors.Wrap(err, "error converting manifest to kube objects")
 	}
 
+	tempManifestBytes, err := yaml.Marshal(manifest)
+	if err != nil {
+		r.log.Error("Failed to marshal temp manifest")
+		return reconcile.Result{}, err
+	}
+
+	tempManifestSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      manifest.GenerateSecretName(manifest.Name),
+			Namespace: request.Namespace,
+		},
+		StringData: map[string]string{
+			"manifest.yaml": string(tempManifestBytes),
+		},
+	}
+
+	// Check if temp manifest secret already exists
+	foundSecret := &corev1.Secret{}
```

Comment:
This entire section shouldn't be required.
What needs to happen:

1. You use the kube converter function (https://github.com/cloudfoundry-incubator/cf-operator/blob/master/pkg/bosh/manifest/kube_converter.go#L56) to generate `ExtendedSecret` definitions
2. You loop through them and create them
3. You create an `ExtendedJob` that uses the secrets *that will be created by the ExtendedSecrets*

You don't need to have logic that tests if the secrets have been created.
You don't need to create the secret that contains the interpolated manifest, the extended job does that on its own.